step-framework-parent/step-framework-timeseries/src/main/java/step/core/timeseries/bucket/BucketBuilder.java

```diff
@@ -58,9 +71,24 @@ public BucketBuilder accumulate(Bucket bucket) {
             bucketDistribution.forEach((key, value) ->
                     distribution.computeIfAbsent(key, k -> new LongAdder()).add(value));
         }
+        accumulateAttributes(bucket);
         return this;
     }
 
+    private void accumulateAttributes(Bucket bucket) {
+        BucketAttributes bucketAttr = bucket.getAttributes();
+        if (accumulateAttributeKeys != null && !accumulateAttributeKeys.isEmpty() && bucketAttr != null && !bucketAttr.isEmpty()) {
+            accumulateAttributeKeys.forEach(a -> {
+                if (bucketAttr.containsKey(a)) {
```

Comment:
The method containsKey retrieves the value in HashMap. It ouwld be therefore more efficient to directly call bucketAttr.get(a) and store it to a variable for line 85 and check if not null

Reply:
Changed as requested, indeed a good perf improvement.